rfhub2/cli/rfhub_importer.py

```diff
@@ -264,3 +317,124 @@ def _has_test_case_table(data: str) -> bool:
+                        new_collection, existing_collection
+                    )
+                    if RfhubImporter._collection_path_and_name_match(
+                        new_collection, reduced_collection
+                    ):
+                        if RfhubImporter._library_or_resource_changed(
+                            new_collection, reduced_collection
+                        ):
+                            collections_to_update.append(new_collection)
+        return collections_to_update
+
+    @staticmethod
+    def _get_new_collections(
+        existing_collections: List[Dict], new_collections: List[Dict]
+    ) -> List[Dict]:
+        """Returns list of collections to insert that were found in paths but not in application"""
+        return [
+            new_collection
+            for new_collection in new_collections
+            if all(
+                new_collection["path"] != existing_collection["path"]
+                for existing_collection in existing_collections
+            )
+        ]
+
+    @staticmethod
+    def _reduce_collection_items(
+        new_collection: Dict, existing_collection: Dict
+    ) -> Dict:
+        reduced_collection = RfhubImporter._get_reduced_collection(
+            new_collection, existing_collection
+        )
+        reduced_collection["keywords"] = RfhubImporter._get_reduced_keywords(
+            new_collection["keywords"], reduced_collection["keywords"]
+        )
+        return reduced_collection
+
+    @staticmethod
+    def _get_reduced_collection(
+        new_collection: Dict, existing_collection: Dict
+    ) -> Dict:
+        """Returns existing_collection dictionary with key/value pairs reduced to the ones from new_collection"""
+        return {k: existing_collection[k] for k in new_collection.keys()}
+
+    @staticmethod
+    def _get_reduced_keywords(
+        new_collection_keywords: List[Dict], existing_collection_keywords: List[Dict]
+    ) -> List[Dict]:
+        if min(len(new_collection_keywords), len(existing_collection_keywords)) > 0:
+            return [
+                {
+                    k: v
+                    for k, v in keyword.items()
+                    if k in new_collection_keywords[0].keys()
+                }
+                for keyword in existing_collection_keywords
+            ]
+
+    @staticmethod
+    def _collection_path_and_name_match(
+        new_collection: Dict, existing_collection: Dict
+    ) -> bool:
+        return (
+            new_collection["name"] == existing_collection["name"]
+            and new_collection["path"] == existing_collection["path"]
+        )
+
+    @staticmethod
+    def _library_or_resource_changed(
+        new_collection: Dict, existing_collection: Dict
+    ) -> bool:
+        if new_collection["type"] == "library":
+            return new_collection["version"] != existing_collection["version"]
+        else:
+            return new_collection != existing_collection
```

Comment:
Way too sensitive. Needs to be changed to first compare resource overview and keywords as list for changes, not as a single item. Ie two `all()`  for twa way compare